molo/profiles/forms.py

```diff
@@ -66,15 +83,17 @@ class EditProfileForm(forms.ModelForm):
         ),
         required=False
     )
+    mobile_number = PhoneNumberField(required=False)
 
     class Meta:
         model = UserProfile
-        fields = ['alias', 'date_of_birth']
+        fields = ['alias', 'date_of_birth', 'mobile_number']
 
     def clean(self):
         alias = self.cleaned_data.get('alias', None)
         date_of_birth = self.cleaned_data.get('date_of_birth', None)
-        if (alias or date_of_birth):
+        mobile_number = self.cleaned_data.get('mobile_number', None)
```

Comment:
Noting one of the things we discussed so I don't forget it later: we need to allow the user to not have to enter a number if `mobile_number_required` is true but `show_mobile_number_field` is false.